lib/OpenQA/WebAPI/Controller/API/V1/Worker.pm

```diff
@@ -136,7 +136,7 @@ sub _incomplete_previous_job {
     my $job_id = $job->id;
     return 0 if $jobs_worker_says_it_works_on->{$job_id};
     my $job_state = $job->state;
-    return 1 if $job_state eq OpenQA::Jobs::Constants::SCHEDULED;
+    return 1 if $job_state eq NEW || $job_state eq SCHEDULED;
```

Comment:
I think in all logic we should avoid "this state || that state" and instead use aggregated states where we would need to check for more than one state. Same as you did in https://github.com/os-autoinst/openQA/pull/6157/files#diff-1c4e8d9bc0096026bc67314675374924c7a5da02359b426a1c6401eac9e90abeR1814